js/manager.js

```diff
@@ -1,149 +1,144 @@
-                if (msg.content.data.method === 'update') {
-                    resolve({comm: comm, msg: msg})
-                }
-            });
-            comm.send({method: 'request_state'}, {})
-        })
-    }
-
-    get onError() {
-        return this._onError
-    }
-
-    registerWithKernel(kernel) {
-        if (this._commRegistration) {
-            this._commRegistration.dispose();
-        }
-        this._commRegistration = kernel.registerCommTarget(
-            this.comm_target_name,
-            (comm, message) =>
-                this.handle_comm_open(new ShimmedComm(comm), message)
-        );
-    }
-
-    display_view(msg, view, options) {
-        const el = options.el || this.el;
-        return Promise.resolve(view).then(view => {
-            pWidget.Widget.attach(view.pWidget, el);
-            view.on('remove', function() {
-                console.log('view removed', view);
-            });
-            return view;
-        });
-    }
-
-    loadClass(className, moduleName, moduleVersion) {
-        if (moduleName === '@jupyter-widgets/output') {
-            return Promise.resolve(outputWidgets).then(module => {
-                if (module[className]) {
-                    return module[className];
-                } else {
-                    return Promise.reject(
-                        `Class ${className} not found in module ${moduleName}`
-                    );
-                }
-            })
-        } else {
-            return super.loadClass(className, moduleName, moduleVersion)
-        }
-    }
-
-    callbacks(view) {
-        const baseCallbacks = super.callbacks(view)
-        return {
-            ...baseCallbacks,
-            iopub: { output: (msg) => this._onError.emit(msg) }
-        }
-    }
-
-    _create_comm(target_name, model_id, data, metadata) {
-        const comm = this.kernel.connectToComm(target_name, model_id)
-        if (data || metadata ) {
-            comm.open(data, metadata)
-        }
-        return Promise.resolve(new ShimmedComm(comm))
-    }
-
-    _get_comm_info() {
-        return this.kernel.requestCommInfo({ target: this.comm_target_name})
-            .then(reply => reply.content.comms)
-    }
-}
+/***************************************************************************
+* Copyright (c) 2018, Voila contributors                                   *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+import { RenderMimeRegistry, standardRendererFactories } from '@jupyterlab/rendermime';
+import { WidgetManager as JupyterLabManager } from '@jupyter-widgets/jupyterlab-manager/lib/manager';

```

Comment:
@jupyter-widgets/jupyterlab-manager does not have an API to consume the manager class directly. 
What approach would work best here ? 
I could always submit a PR there and see if they want to expose the manager ?

Reply:
cc @jasongrout I think that this could be a meaningful change for the next release of the JupyterLab manager.